connector/spanmetricsconnector/connector.go

```diff
@@ -65,6 +72,9 @@ type connectorImp struct {
 	histograms metrics.HistogramMetrics
 	sums       metrics.SumMetrics
 
+	// unit divider, used to convert nanoseconds to milliseconds or seconds
+	unitDivider int64
```

Comment:
This is only read once [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/19370/files#diff-4cf1d004574eff4ea9a26ff033740136245993c942742b4e8c6622e522599910R298). Since we have the information needed to derive a `unitDivider` via `connectorImp.config.Histogram.Unit`, what do you think about removing this and just mapping the `unitDiver` value again when it's needed?

Reply:
Then we would have to do the lookup in the map and call the func derived from map for each span.

Reply:
I think it's still a constant time operation; but I'm not strongly attached to my view, so I'm happy for you to leave the field as is.

Reply:
I'll drop it.